regulations/tests/layers_formatting_tests.py

```diff
@@ -132,3 +132,36 @@ def ret_mock(arg):
         context = render.call_args[0][0]
         self.assertEqual(context['variable'], 'abc')
         self.assertEqual(context['subscript'], '123')
+
+
+    @patch('regulations.generator.layers.formatting.loader')
+    def test_apply_layer_dash(self, loader):
+        mocks = {'table': Mock(), 'note': Mock(), 'code': Mock(),
+                 'subscript': Mock(), 'dash': Mock()}
+
+        def ret_mock(arg):
+            for key in mocks:
+                if key in arg:
+                    return mocks[key]
+
+        loader.get_template.side_effect = ret_mock
+        render = mocks['dash'].render
```

Comment:
Yes, I thought these looked really hokey too. I like your suggestion; I don't think there's a better way to handle a conditional side effect than a function, and a lambda here is far more concise. Done and done!